cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/AbstractStreamService.java

```diff
@@ -16,14 +16,17 @@
 
 package co.cask.cdap.data.stream.service;
 
+import co.cask.cdap.common.conf.Constants;
 import co.cask.cdap.data.stream.StreamCoordinatorClient;
-import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.common.util.concurrent.AbstractScheduledService;
 import com.google.common.util.concurrent.Service;
 
+import java.util.concurrent.TimeUnit;
+
 /**
  * Stream service meant to run in an HTTP service.
  */
-public abstract class AbstractStreamService extends AbstractIdleService implements StreamService {
+public abstract class AbstractStreamService extends AbstractScheduledService implements StreamService {
```

Comment:
When extends from the `AbstractScheduledService`, it's good to override the `executor()` method to control the executor and the shutdown of it. See `AbstractResourceReporter` as an example.